usecases/backup/backend.go

```diff
@@ -122,20 +125,78 @@ func (s *objStore) meta(ctx context.Context, key string, dest interface{}) error
 }
 
 type nodeStore struct {
-	objStore
+	objectStore
+}
+
+// https://github.com/weaviate/weaviate/pull/6012#pullrequestreview-2371394228
+type NodeStore interface {
+	GetObjectStore(ctx context.Context, key string, overrideBucket, overridePath string) *objectStore
+	SetObjectStore(os *objectStore)
+	GetBackend() modulecapabilities.BackupBackend
+	SetBackend(backend modulecapabilities.BackupBackend)
+	GetBackupId() string
+	SetBackupId(backupId string)
+	GetBucket() string
+	SetBucket(bucket string)
+	GetPath() string
+	SetPath(path string)
```

Comment:
I think that we should remove those `Get*` and `Set*` methods, they are not used anywhere and they enable users to modify internal private variables which should be immutable

Reply:
done